server/redis.repository.js

```diff
@@ -108,4 +111,42 @@ const swipingCompleted = (total_swipes, num_guests, num_movies) => {
     }
 }
 
-module.exports = { initializeRoom, likeEvent, dislikeEvent, purgeRoom };
+const checkCache = async (requestUrl) => {
+    try {
+        return await existsAsync(requestUrl);
+    } catch (error) {
+        return error;
+    }
+}
+
+/**
+ * This function should only be called AFTER checkCache() is called with a response of 1 (key exists)
+ * @param {String} requestUrl literal request URL that corresponds with a cached redis key
+ * @returns {Object} Returns JSON string of cached response
+ */
+const getCachedCall = async (requestUrl) => {
+    try {
+        return await getAsync(requestUrl);
+    } catch (error) {
+        return error
+    }
+}
+    
+
+/**
+ * 
+ * @param {String} requestUrl literal request URL that corresponds with a cached redis key
+ * @param {String} data response data in the form of a JSON string to be used later
+ * @returns {Integer} number of fields that were added
```

Comment:
Will this number always be 1? If so, we should note that we just expect it to return 1 in the method signature here.

Reply:
Thats right, the functionality as we have it right now should only result in a return value of 1 since we are only adding one key:value pair to redis. I will update the method signature to reflect that